components/post_view/acknowledgements/post_acknowledgements.tsx

```diff
@@ -110,20 +120,20 @@ function PostAcknowledgements({
                 className={classNames({
                     AcknowledgementButton: true,
                     'AcknowledgementButton--acked': Boolean(acknowledgedAt),
-                    'AcknowledgementButton--disabled': Boolean(acknowledgedAt) && moreThan5minAgo(acknowledgedAt),
+                    'AcknowledgementButton--disabled': buttonDisabled,
                     'AcknowledgementButton--default': !list || list.length === 0,
                 })}
                 {...getReferenceProps()}
             >
                 <CheckCircleOutlineIcon size={16}/>
-                {(list && list.length > 0) ? list!.length : (
+                {(list && list.length > 0) || isCurrentAuthor ? list?.length || 0 : (
```

Comment:
1/5 - You might want to pull this out of the JSX and/or declare it separately with an if-else for each case. The mix of boolean logic, a ternary, and then more boolean logic for handling `list` being undefined makes this a bit difficult to read, especially as part of a JSX block